test/e2e/gitops_service_test.go

```diff
@@ -548,6 +549,97 @@ func validateNonDefaultArgocdNamespaceManagement(t *testing.T) {
 
 }
 
+func validateGrantingPermissionsByLabel(t *testing.T) {
+	framework.AddToFrameworkScheme(argoapi.AddToScheme, &argoapp.ArgoCD{})
+	ctx := framework.NewContext(t)
+	cleanupOptions := &framework.CleanupOptions{TestContext: ctx, Timeout: time.Second * 60, RetryInterval: time.Second * 1}
+	defer ctx.Cleanup()
+	f := framework.Global
+
+	sourceNS := "source-ns"
+	argocdInstance := "argocd-label"
+
+	// create a new source namespace
+	sourceNamespaceObj := &corev1.Namespace{
+		ObjectMeta: v1.ObjectMeta{
+			Name: sourceNS,
+		},
+	}
+	err := f.Client.Create(context.TODO(), sourceNamespaceObj, cleanupOptions)
+	if !kubeerrors.IsAlreadyExists(err) {
+		assertNoError(t, err)
+	}
+
+	// create an ArgoCD instance in the source namespace
+	argoCDInstanceObj := &argoapp.ArgoCD{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      argocdInstance,
+			Namespace: sourceNS,
+		},
+	}
+	err = f.Client.Create(context.TODO(), argoCDInstanceObj, cleanupOptions)
+	assertNoError(t, err)
+
+	// create a target namespace to deploy resources
+	// allow argocd to create resources in the target namespace by adding managed-by label
+	targetNS := "target-ns"
+	targetNamespaceObj := &corev1.Namespace{
+		ObjectMeta: v1.ObjectMeta{
+			Name: sourceNS,
+			Labels: map[string]string{
+				"argocd.argoproj.io/managed-by": sourceNS,
+			},
+		},
+	}
+	err = f.Client.Create(context.TODO(), targetNamespaceObj, cleanupOptions)
+	if !kubeerrors.IsAlreadyExists(err) {
+		assertNoError(t, err)
+	}
+
+	// check if the necessary roles/rolebindings are created in the target namespace
+	resourceList := []resourceList{
+		{
+			&rbacv1.Role{},
+			[]string{
+				argocdInstance + "-argocd-application-controller",
+				argocdInstance + "-argocd-redis-ha",
+				argocdInstance + "-argocd-server",
+			},
+		},
+		{
+			&rbacv1.RoleBinding{},
+			[]string{
+				argocdInstance + "-argocd-application-controller",
+				argocdInstance + "-argocd-redis-ha",
+				argocdInstance + "-argocd-server",
+			},
+		},
+	}
+	err = waitForResourcesByName(resourceList, targetNS, time.Second*60, t)
+	assertNoError(t, err)
+
+	// create an ArgoCD app and check if it can create resources in the target namespace
+	taxiAppCr := filepath.Join("test", "appcrs", "taxi_appcr.yaml")
+	ocPath, err := exec.LookPath("oc")
+	assertNoError(t, err)
+	cmd := exec.Command(ocPath, "apply", "-f", taxiAppCr)
+	err = cmd.Run()
+	assertNoError(t, err)
+
+	err = wait.Poll(time.Second*1, time.Second*60, func() (bool, error) {
+		if err := helper.ApplicationHealthStatus("taxi", sourceNS); err != nil {
```

Comment:
@chetan-rns Shouldn't we be deploying the application in the target namespace and checking for it there rather than the source namespace?

Reply:
Argo CD can only manage applications in the namespace the instance is running, which is the source namespace.

Reply:
Isn't the idea that through this feature you'd be able to deploy applications in other namespaces that would be managed by the argocd-instance through the label? 
IIRC that is also what Shubham did in his demo

Reply:
Every non-default Argo CD instance will have admin rights in its namespace( the namespace where it's installed). So an instance running in NS-A won't have enough permissions to create k8s resources in NS-B. We can extend the access by explicitly creating a role/binding in NS-B granting access to Argo CD. Shubham's PR automates this step by allowing users to add labels and the operator would take care of roles/bindings behind the scene. The application CR would still reside in NS-A where Argo CD is running. In this context, the application refers to the Argo CD application CR that deploys k8s resources to the target namespace NS-B.

Reply:
@jaideepr97, I think what Chetan's trying to say here is that the Application CR will reside in the `source-ns` while the application would be deployed in `target-ns` which is how the application deployment in ArgoCD works. So I think that should be fine !!

Reply:
Oh got it, thanks @chetan-rns @shubhamagarwal19